### PR TITLE
feat: schema contact api endpoint

### PR DESCRIPTION
**Pull request by connoratrug**
Add endpoint to send message

endpoint: `/:schema/api/message/`

Api accepts a post request with application/json body, json object must contain:
```
{
    recipientsFilter: string,
    subject: string,
    body: string"
}
```

example: 
```
{
    "recipientsFilter": "{\"filter\": {\"name\":{\"equals\":\"Test cohort\"}}}",
    "subject": "Say hello to my catalogue",
    "body": "Just wanted to say hello"
}
```

**recipientsFilter**: should be a valid emx2 gql filter
**subject**: human readable message subject
**body**: the message that needs to be send

**Message recipients**

The message receivers a determined by executing a query on the schema the resulting tree is flattened and all leaves are used as message recipients.

The query can be set via ```contactRecipientsQuery``` 
 for example: 

```
query Resources($filter:ResourcesFilter){ 
    Resources(filter: $filter) { 
        contacts { 
            email 
        }
   }
 }
```

To test; set env vars and run the send mail test ( use gmail for single test). Molgenis prod setup works via internal network ( no auth, default settings )

### CODE REVIEW

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlSchema.java [hunk 1]
```diff
@@ -402,6 +402,19 @@ public Integer getChangesCount() {
     return metadata.getChangesCount();
   }
 
+  @Override
+  public String getSettingValue(String key) {
+    String setting = metadata.getSetting(key);
+    if (setting == null) {
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlSchema.java:408]: should we not throw already in the metadata?

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MessageApi.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    final Boolean sendResult =
+        emailService.send(recipients, sendMessageAction.subject(), sendMessageAction.body());
+    return sendResult.toString();
+  }
+
+  private static EmailSettings loadEmailSettings(Schema schema) {
+    EmailSettings.EmailSettingsBuilder builder = new EmailSettings.EmailSettingsBuilder();
+
+    setEmailSetting(schema, "EMAIL_HOST", builder::host);
```
Comment (mswertz) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MessageApi.java:97]: for now okay, but I kinda feel the need to convert this into one setting of type json and then have validation on that json schema. And if we play it smartly we could use a emx2 metadata schema to manage such json object using standard RowEdit user interface forms (as if it was a database row)?

**Review by mswertz — COMMENTED** [review]
code looks great, just some minor considerations.
Only thing I would like changed is to move the path mapping into create() method just with all other API. I understood you had some strange issues with it somehow being hidden by other api?

@mswertz i had that but then moved it into the server setup, the order and matching of the combined paths matters, therefore it is better to have them in a single place ( it's like changing the global ( path) scope). otherwise you need to jump into every create function to figure out what is path setup is.

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MessageApi.java (hunk 2 above)
Comment (connoratrug) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MessageApi.java:97]: thought about doing this initially , but think using simple key val fields  is probably a better solution, if we want we can add a simple email setting tab to the ui and configure a form to show inputs ( and add validation and stuff). why a user type json and make "" error when we can have them add a string ( of add a form )

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisWebservice.java [hunk 3]
```diff
@@ -52,6 +52,8 @@ public static void start(int port) {
      * WARNING !! SPARK JAVA USES DESIGN WHERE THE ORDER OF REQUEST DEFINITION DETERMINES THE HANDLER
      */
 
+    post("/:schema/api/message/*", ACCEPT_JSON, MessageApi::send);
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisWebservice.java:55]: i really prefer to have it  here, i initially spend 4 hours trying to figure out why i could pass the body to the endpoint (  tuned out it was being filtered by a different path)

File: docs/molgenis/dev_messageapi.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Message api 
+
+Endpoint to send message
+
+endpoint: `/:schema/api/message/`
+
+Api accepts a post request with application/json body, json object must contain:
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply docs/molgenis/dev_messageapi.md:7]: not a fan of using gql endpoints for actions / method calls, just do not be the advantage ( disadvantage is more complex call, smaller set of dev familiar with gql)

File: docs/molgenis/dev_messageapi.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    recipientsFilter: string,
+    subject: string,
+    body: string"
+}
+```
+
+example:
+```
+{
+    "recipientsFilter": "{\"filter\": {\"name\":{\"equals\":\"Test cohort\"}}}",
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply docs/molgenis/dev_messageapi.md:19]: personally not sure about all the {{ and "" and \\\ but still thing is better then adding a second filter syntax

**Review by mswertz — APPROVED** [review]

**Pull request merged**